micromamba.el

```diff
@@ -158,6 +166,62 @@ Returns an alist with the following keys:
       (vars-export . ,vars-export)
       (scripts . ,scripts))))
 
+(defun micromamba--find-env-yml (dir) ;; adapted from conda.el
+  "Find an environment.yml or .yaml in DIR or its parent directories."
+  (let ((containing-path (locate-dominating-file dir
+                          (lambda (parent)
+                            (directory-files parent nil "environment.[yml|yaml]")))))
+    (when containing-path
+      (let ((yml-candidate
+             (concat (file-name-as-directory containing-path) "environment.yml"))
+            (yaml-candidate
+             (concat (file-name-as-directory containing-path) "environment.yaml")))
+        (or (when (file-readable-p yml-candidate) yml-candidate)
+            (when (file-readable-p yaml-candidate) yaml-candidate))))))
+
+(defun micromamba--read-file-into-string (filename)
+  "Read the contents of FILENAME into a string."
+  (with-temp-buffer
+    (let ((coding-system-for-read 'utf-8))
+      (insert-file-contents filename)
+      (buffer-string))))
+
+(defun micromamba--get-name-from-env-yml-contents (env-yml-contents)
+  "Pull the `name` property out of a stringified YAML file"
+  (save-match-data
+    (when (string-match "name:[ ]*\\([A-z0-9-_.]+\\)[ ]*$" env-yml-contents)
+      (match-string 1 env-yml-contents))))
+
+(defun micromamba--get-name-from-env-yml (filename) ;; adapted from conda.el
+  "Pull the `name` property out of the YAML file at FILENAME."
+  (when filename
+    (let ((filename (file-truename filename)))
+      (or (unless (file-has-changed-p filename))
+          (cdr (assoc filename micromamba--yml-cache))
+          (setf (alist-get filename micromamba--yml-cache)
```

Comment:
This won't reliably work because the default test function of `alist-get` is `eq`, which is merely an identity comparison. That is, it returns `t` when the operands are the same object, not when they are two different objects with the same content, which strings might be.

```lisp
(setq foo "bar")

(eq foo foo) ; => t
(eq foo "bar") ; => nil
(eq "bar" "bar"); => nil
(equal foo "bar"); => t
```

So this line needs to be:
```suggestion
          (setf (alist-get filename micromamba--yml-cache nil nil #'equal)
```
This looks a bit hacky, but I haven't found a better way for alists with string keys.